src/transformers/integrations/bitsandbytes.py

```diff
@@ -74,32 +79,55 @@ class `Int8Params` from `bitsandbytes`.
 
             # Support models using `Conv1D` in place of `nn.Linear` (e.g. gpt2) by transposing the weight matrix prior to quantization.
             # Since weights are saved in the correct "orientation", we skip transposing when loading.
-            if issubclass(module.source_cls, Conv1D) and fp16_statistics is None:
+            if issubclass(module.source_cls, Conv1D) and not prequantized_loading:
                 new_value = new_value.T
 
             kwargs = old_value.__dict__
+
+            assert prequantized_loading == (
+                new_value.dtype in (torch.int8, torch.uint8)
+            ), f"Value dtype `{new_value.dtype}` is not compatible with parameter quantization status."
+
             if is_8bit:
+                is_8bit_serializable = version.parse(importlib.metadata.version("bitsandbytes")) > version.parse(
+                    "0.37.2"
+                )
+                if new_value.dtype in (torch.int8, torch.uint8):
+                    if not is_8bit_serializable:
+                        raise ValueError(
+                            "Detected int8 weights but the version of bitsandbytes is not compatible with int8 serialization. "
+                            "Make sure to download the latest `bitsandbytes` version. `pip install --upgrade bitsandbytes`."
+                        )
                 new_value = bnb.nn.Int8Params(new_value, requires_grad=False, **kwargs).to(device)
+                if prequantized_loading:
+                    setattr(new_value, "SCB", fp16_statistics.to(device))
+
             elif is_4bit:
-                new_value = bnb.nn.Params4bit(new_value, requires_grad=False, **kwargs).to(device)
+                if prequantized_loading:
+                    is_4bit_serializable = version.parse(importlib.metadata.version("bitsandbytes")) > version.parse(
+                        "0.41"
+                    )
+                    # TODO update version number after BNB release with PR #753
+                    if not is_4bit_serializable:
+                        raise ValueError(
+                            "Detected 4-bit weights but the version of bitsandbytes is not compatible with 4-bit serialization. "
+                            "Make sure to download the latest `bitsandbytes` version. `pip install --upgrade bitsandbytes`."
+                        )
+                    new_value = bnb.nn.Params4bit.from_prequantized(
+                        data=new_value, quantized_stats=quantized_stats, requires_grad=False, device=device, **kwargs
+                    )
+                else:
+                    new_value = bnb.nn.Params4bit(new_value, requires_grad=False, **kwargs).to(device)
 
             module._parameters[tensor_name] = new_value
-            if fp16_statistics is not None:
-                setattr(module.weight, "SCB", fp16_statistics.to(device))
+
+        else:
+            pass  # param.device.type == "cuda"  # is it a possible loading scenario?
```

Comment:
```suggestion
```
Hmm I don't really understand this 🤔 Can you elaborate more ? I suspect it is not needed so I propose to remove it

Reply:
since the preceding branch above is `if param.device.type != "cuda":` I thouhgt that the opposite should be handled too. I changed this code to raise `ValueError`.